opennms-services/src/main/java/org/opennms/netmgt/notifd/ServiceRegistryExecutor.java

```diff
@@ -28,25 +28,28 @@
 
 package org.opennms.netmgt.notifd;
 
+import java.lang.management.ManagementFactory;
+import java.util.List;
+
+import org.opennms.core.soa.lookup.ServiceLookup;
+import org.opennms.core.soa.lookup.ServiceLookupBuilder;
 import org.opennms.core.soa.support.DefaultServiceRegistry;
 import org.opennms.netmgt.model.notifd.Argument;
 import org.opennms.netmgt.model.notifd.NotificationStrategy;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.lang.management.ManagementFactory;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-
 public class ServiceRegistryExecutor implements ExecutorStrategy {
     private static final Logger LOG = LoggerFactory.getLogger(ServiceRegistryExecutor.class);
 
-    private static final DefaultServiceRegistry s_registry = DefaultServiceRegistry.INSTANCE;
-
     private static final String GRACE_PERIOD_MS_SYS_PROP = "org.opennms.netmgt.notifd.notificationStrategyGracePeriodMs";
     private static final int GRACE_PERIOD_MS = Integer.getInteger(GRACE_PERIOD_MS_SYS_PROP, 3*60*1000);
     private static final int LOOKUP_DELAY_MS = 5*1000;
 
+    private static final ServiceLookup SERVICE_LOOKUP = new ServiceLookupBuilder(DefaultServiceRegistry.INSTANCE)
+            .blocking(GRACE_PERIOD_MS, LOOKUP_DELAY_MS, () -> ManagementFactory.getRuntimeMXBean().getUptime())
```

Comment:
Can we make `ManagementFactory.getRuntimeMXBean().getUptime()` the default uptime provider instead of having to pass this each time?

With the properties moved into the class, the a default uptime provider, most of the calls could just end up looking like:
```
new ServiceLookupBuilder(DefaultServiceRegistry.INSTANCE)
  .blocking()
  .build();
```

Reply:
Yep, makes also sense.